modules/analysis-common/src/main/java/org/opensearch/analysis/common/HyphenationCompoundWordTokenFilterFactory.java

```diff
@@ -54,11 +54,16 @@
  */
 public class HyphenationCompoundWordTokenFilterFactory extends AbstractCompoundWordTokenFilterFactory {
 
+    private final boolean noSubMatches;
+    private final boolean noOverlappingMatches;
     private final HyphenationTree hyphenationTree;
 
     HyphenationCompoundWordTokenFilterFactory(IndexSettings indexSettings, Environment env, String name, Settings settings) {
         super(indexSettings, env, name, settings);
 
+        noSubMatches = settings.getAsBoolean("no_sub_matches", false);
+        noOverlappingMatches = settings.getAsBoolean("no_overlapping_matches", false);
```

Comment:
I am wondering if the settings should have more qualified path?

Reply:
@jainankitk 
could you please add more details on this recommendation 
kinda new here! :smiley:

Reply:
As discussed during community meeting, can you share the json for providing this setting?

Reply:
I added a bunch of tests that should show the settings